kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/data/ColumnVectorConverter.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.data;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.types.*;
+import io.delta.kernel.defaults.internal.data.vector.*;
+
+public class ColumnVectorConverter {
+    public static ColumnVector convertToOutputType(ColumnVector c, DataType outputType) {
+        DataType inputType = c.getDataType();
+        if (inputType == outputType) {
+            return c;
+        }
+        if (outputType instanceof BooleanType) {
+            // TODO: how to convert string to boolean?
+            // "true" "TRUE", "T" -> true ?
```

Comment:
I couldn't find info on how to resolve string to boolean https://spark.apache.org/docs/latest/sql-ref-datatypes.html 
So I am making the least common type resolution throwing an error temporary.